javascript/extractor/src/com/semmle/js/extractor/FileExtractor.java

```diff
@@ -511,7 +511,7 @@ private Integer extractContents(
           new TextualExtractor(
               trapwriter, locationManager, source, config.getExtractLines(), metrics, extractedFile);
       LoCInfo loc = extractor.extract(textualExtractor);
-      int numLines = textualExtractor.getNumLines();
+      int numLines = textualExtractor.isSnippet() ? 0 : textualExtractor.getNumLines();
       int linesOfCode = loc.getLinesOfCode(), linesOfComments = loc.getLinesOfComments();
       trapwriter.addTuple("numlines", fileLabel, numLines, linesOfCode, linesOfComments);
```

Comment:
Lets me see if I'm understanding this right. 

The reason you set `numLines` to 0 is because `fileLabel` refers to the source-file (that contains the snippet), while `numLines` contains the number of lines in the snippet itself?  

And a tuple for `numLines` with the correct number will be added when the source-file is extracted?

Wouldn't it then be better to skip the `"numlines"` and `"filetype"` tuples if we are extracting a snippet? 

That way you can also revert the `int getNumberOfLines() { result = sum(int loc | numlines(this, loc, _, _) | loc) }` you are doing in `Files.qll`.

Reply:
> The reason you set numLines to 0 is because fileLabel refers to the source-file (that contains the snippet), while numLines contains the number of lines in the snippet itself?

Yes

> And a tuple for numLines with the correct number will be added when the source-file is extracted?

A tuple consists of three numbers: total 'physical' lines, lines of code, and lines of comments.

The 'physical' lines is easiest to populate when processing the original file, but the latter two aren't known at that time. There is no point in time at which all three numbers are known simultaneously, unless we start adding more extractor state.

So I made the rule that the columns of `numlines` must be summed up. It seemed like the most honest way to model the fact that a single file is the result of multiple TRAP files.